### DISCUSSION TITLE
An error occurs when invoking LAMMPS in parallel within the DeepMD environment.

### DISCUSSION THREAD

**Discussion by niuniu800** [deepmodeling/deepmd-kit#4121] [Q&A]
software version:
deepmd-kit  2.2.3,dpgen  0.12.1,tensorflow 2.9.0.
When submitting the LAMMPS job using the submission script, the following error occurred.
![Uploading err.png…]()
submission script:
#!/bin/bash
#PBS -N bn1639
#PBS -o o.out
#PBS -e e.err
#PBS -l nodes=1:ppn=2
#PBS -l walltime=3600:00:00
#source /opt/intel/oneapi/intelpython/latest/etc/profile.d/conda.sh
source activate  /home/hanye/.conda/envs/deepmd-kit
export PATH=$HOME/bin:$PATH

#source ~/.bashrc
#source /opt/intel/oneapi/setvars.sh

#ulimit -s unlimited
#ulimit -c unlimited

NP=$(cat ${PBS_NODEFILE} | wc -l)

cd $PBS_O_WORKDIR

mpirun -np 2 ~/.conda/envs/deepmd-kit/bin/lmp -i in.rlx >my.out
But I can run it directly by typing mpirun -np 1 ~/.conda/envs/deepmd-kit/bin/lmp -i in.rlx >my.out in the command line.How should this problem be resolved?

**Comment by niuniu800** [comment]
e.err file:
/home/hanye/.bashrc: line 25: ulimit: stack size: cannot modify limit: Operation not permitted
DeePMD-kit WARNING: Environmental variable TF_INTRA_OP_PARALLELISM_THREADS is not set. Tune TF_INTRA_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable TF_INTER_OP_PARALLELISM_THREADS is not set. Tune TF_INTER_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable OMP_NUM_THREADS is not set. Tune OMP_NUM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable TF_INTRA_OP_PARALLELISM_THREADS is not set. Tune TF_INTRA_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable TF_INTER_OP_PARALLELISM_THREADS is not set. Tune TF_INTER_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable OMP_NUM_THREADS is not set. Tune OMP_NUM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable TF_INTRA_OP_PARALLELISM_THREADS is not set. Tune TF_INTRA_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable TF_INTER_OP_PARALLELISM_THREADS is not set. Tune TF_INTER_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable OMP_NUM_THREADS is not set. Tune OMP_NUM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
2024-09-12 09:51:46.535774: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE4.1 SSE4.2 AVX AVX2 AVX512F AVX512_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-09-12 09:51:46.536527: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE4.1 SSE4.2 AVX AVX2 AVX512F AVX512_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-09-12 09:51:46.538726: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.
2024-09-12 09:51:46.539596: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.
2024-09-12 09:51:46.556748: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:354] MLIR V1 optimization pass is not enabled
2024-09-12 09:51:46.556976: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:354] MLIR V1 optimization pass is not enabled
/home/hanye/.conda/envs/deepmd-kit/bin/lmp: line 11: 439787 Floating point exception(core dumped) /home/hanye/.conda/envs/deepmd-kit/bin/_lmp "$@"

my.out file:

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 63 PID 440369 RUNNING AT hanye64
=   KILLED BY SIGNAL: 9 (Killed)
===================================================================================

**Comment by njzjz** [comment]
It has been fixed in v2.2.7.

**Reply by niuniu800** [reply]
Thank you for your response! So, upgrading deepmd-kit to v2.2.7 is all that's needed to resolve the issue?

**Reply by njzjz** [reply]
It was fixed by #2923. Or you can prevent zero atoms in a processor, which is meaningless for parallelism.
